pkg/crd/persistentip/v1alpha1/types.go

```diff
@@ -0,0 +1,48 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//go:generate go run sigs.k8s.io/controller-tools/cmd/controller-gen@v0.13.0 paths=./... object output:dir=./
+
+//go:generate go run k8s.io/code-generator/cmd/client-gen@v0.28.0 client-gen --go-header-file ../../../../hack/custom-boilerplate.go.txt --clientset-name versioned --input-base "" --input github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1 --output-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/clientset ..
+
+//go:generate go run k8s.io/code-generator/cmd/lister-gen@v0.28.0 lister-gen --go-header-file ../../../../hack/custom-boilerplate.go.txt --input-dirs github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1 --output-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/listers ..
+
+//go:generate go run k8s.io/code-generator/cmd/informer-gen@v0.28.0 informer-gen --go-header-file ../../../../hack/custom-boilerplate.go.txt --input-dirs github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1 --versioned-clientset-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/clientset/versioned --listers-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/listers --output-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/informers ..
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=persistentips,shortName=pip,singular=persistentip,scope=Namespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PersistentIP is the Schema for the PersistentIP API
+type PersistentIP struct {
```

Comment:
What do you think about make it plural ? since this can containes multiple IPs
so it would be like 
```golang
// +genclient
// +kubebuilder:object:root=true
// +kubebuilder:resource:path=persistentipses,shortName=pips,singular=persistentips,scope=Namespaced
// +kubebuilder:storageversion
// +kubebuilder:subresource:status
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object

// PersistentIP is the Schema for the PersistentIP API
type PersistentIPs struct {
```

Reply:
Uff .. 

I honestly don't like it, and it doesn't sound like correct English to me.

Making the "short name" plural *does* make sense IMO. Good catch there.

Reply:
I mean that the golang type should be `PersistentIPs` since we have multiple ips per network/interface and on extension we should reflect that at CRDs names, an alternative is name this PersistentAddress and it can contain multiple IPs

Reply:
So make sense to name this `PersistentIPs` ?

Reply:
Well seems you were right. The code generator doesn't let me change the struct to be named `PersistentIPs` but leave the singular / path as they were.

We have to discuss this. IMO, it reads a bit weird. Never the less, given our premature this all is, we can jsut move forward with this, and fix later, when we provide a go releaser configuration - i.e. after having the "approved" CRD.

Reply:
Just so we have somewhere, I did manage to have `PersistentIPs` with the following

```golang
package v1                                                                      
                                                                                
import (                                                                        
    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"                               
)                                                                               
                                                                                
// +genclient                                                                   
// +kubebuilder:object:root=true                                                
// +kubebuilder:resource:path=persistentipses,shortName=pips,singular=persistentips,scope=Namespaced
// +kubebuilder:storageversion                                                  
// +kubebuilder:subresource:status                                              
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object          
// PersistentIPs is the Schema for the OverlappingRangeIPReservations API       
type PersistentIPs struct {                                                     
    metav1.TypeMeta   `json:",inline"`                                          
    metav1.ObjectMeta `json:"metadata,omitempty"`                               
                                                                                
    Spec   PersistentIPsSpec   `json:"spec,omitempty"`                          
    Status PersistentIPsStatus `json:"status,omitempty"`                        
}                                                                               
                                                                                
type PersistentIPsSpec struct {                                                 
    Network string `json:"network"`                                             
    // +optional                                                                
    Interface string `json:"interface,omitempty"`                               
}                                                                               
                                                                                
type PersistentIPsStatus struct {                                               
    IPs []string `json:"ips"`                                                   
}                                                                               
                                                                                
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object          
// +resource:path=persistentipses                                               
type PersistentIPsList struct {                                                 
    metav1.TypeMeta `json:",inline"`                                            
    metav1.ListMeta `json:"metadata,omitempty"`                                 
    Items           []PersistentIPs `json:"items"`                              
}                                                                        
```

Reply:
How about `PersistentAllocation` / `PersistentIPAMAllocation` ?

Reply:
`PersistentIPAMAllocation` That's much better, we may end up storing stuff beyond IPs

Reply:
I'm unsure what's the use case for non persistent ... AFAICT that will tie the object to the lifecycle of the pod .

... which you get when you implement a CNI plugin CNI ADD / DEL. (plus reconcile)

Having said that, that's a bit what was being asked from whereabouts in [this issue](https://github.com/k8snetworkplumbingwg/whereabouts/issues/51#issue-663770280).

Reply:
Sticking w/ `PersistentIPAMAllocations` (plural). The singular would be `PersistentIPAMAllocation`.